client/state/sites/plans/selectors/calculate-plan-upgrade-credits.ts

```diff
@@ -0,0 +1,38 @@
+import { getPlanDiscountedRawPrice } from './get-plan-discounted-raw-price';
+import { getSitePlanRawPrice } from './get-site-plan-raw-price';
+import isPlanAvailableForPurchase from './is-plan-available-for-purchase';
+import type { IAppState } from 'calypso/state/types';
+
+/**
+ * Calculate available plan credits given a set of displayed plans
+ *
+ * @param {*} state app state
+ * @param {*} siteId the selected site id
+ * @param {string[]}  consideredPlans Plans that are considered for the given calcuation
+ * @returns {number} The amount of credits available for the given plans
+ */
+export function calculatePlanUpgradeCredits(
```

Comment:
I was thinking since it is a literal calculation of state to have it closer to redux state so that any future uses of this can reuse this, but looking back there is a possibility for code YAGNI. But the code has to live somewhere so this is more a matter of cohesion I guess. 

The other reason I didn't really move to a hook is because hook semantics like useState or useEffect are not required here and the only use would be to tap into state.

Given the direction of moving away from redux maybe its a good idea to move away from the state and probably rely on a hook and props. Let me know what you prefer, I am ok either way. 👍 

For now I will rename this to adhere with the naming convention.

Reply:
> Given the direction of moving away from redux maybe its a good idea to move away from the state and probably rely on a hook and props. Let me know what you prefer, I am ok either way. 👍

Technically speaking, this component (and selector) - the stuff in the entire PR - will probably not be subject to export to a package (like we want to do with the main grid component). So it seems ok to be utilizing Calypso state selectors here. The refactor you did here is pretty powerful in that regard, as it pulls out a big chunk of code that we don't need to "serve" the 2023 grid from a package 👍 

My suggestion for a hook is to avoid adding a selector that's only used for a specific use case (I think hooks can serve that purpose)